docs/draft/6-payloads.md

```diff
@@ -0,0 +1,359 @@
+If a signature is not present but an author is provided by a layer below, the message is not to be relayed to third parties and it is considered plausibly deniable.
+
+## Encoding
+
+The payload is encoded using [Protobuf](https://developers.google.com/protocol-buffers)
+
+## Types of messages
+
+### Message
+
+The type `ChatMessage` represents a chat message exchanged between clients.
+
+#### Payload
+
+The protobuf description is:
+
+```protobuf
+message ChatMessage {
+  // Lamport timestamp of the chat message
+  uint64 clock = 1;
+  // Unix timestamps in milliseconds, currently not used as we use whisper as more reliable, but here
+  // so that we don't rely on it
+  uint64 timestamp = 2;
+  // Text of the message
+  string text = 3;
+  // Id of the message that we are replying to
+  string response_to = 4;
+  // Ens name of the sender
+  string ens_name = 5;
+  // Chat id, this field is symmetric for public-chats and private group chats,
+  // but asymmetric in case of one-to-ones, as the sender will use the chat-id
+  // of the received, while the receiver will use the chat-id of the sender.
+  // Probably should be the concatenation of sender-pk & receiver-pk in alphabetical order
+  string chat_id = 6;
+
+  // The type of message (public/one-to-one/private-group-chat)
+  MessageType message_type = 7;
+  // The type of the content of the message
+  ContentType content_type = 8;
+
+  oneof payload {
+    StickerMessage sticker = 9;
+    ImageMessage image = 10;
+  }
+
+  enum MessageType {
+    UNKNOWN_MESSAGE_TYPE = 0;
+    ONE_TO_ONE = 1;
+    PUBLIC_GROUP = 2;
+    PRIVATE_GROUP = 3;
+    // Only local
+    SYSTEM_MESSAGE_PRIVATE_GROUP = 4;}
+  enum ContentType {
+    UNKNOWN_CONTENT_TYPE = 0;
+    TEXT_PLAIN = 1;
+    STICKER = 2;
+    STATUS = 3;
+    EMOJI = 4;
+    TRANSACTION_COMMAND = 5;
+    // Only local
+    SYSTEM_MESSAGE_CONTENT_PRIVATE_GROUP = 6;
+    IMAGE = 7;
+  }
+}
+```
+
+#### Payload
+
+| Field | Name | Type | Description |
+| ----- | ---- | ---- | ---- |
+| 1 | clock | `uint64` | The clock of the chat|
+| 2 | timestamp | `uint64` | The sender timestamp at message creation |
+| 3 | text | `string` | The content of the message |
+| 4 | response_to | `string` | The ID of the message replied to |
+| 5 | ens_name | `string` | The ENS name of the user sending the message |
+| 6 | chat_id | `string` | The local ID of the chat the message is sent to |
+| 7 | message_type | `MessageType` | The type of message, different for one-to-one, public or group chats |
+| 8 | content_type | `ContentType` | The type of the content of the message | 
+| 9 | payload | `Sticker|Image|nil` | The payload of the message based on the content type |
+
+#### Content types
+
+Content types are required for a proper interpretation of incoming messages. Not each message is plain text but may carry a different information.
+
+The following content types MUST be supported:
+* `TEXT_PLAIN` identifies a message which content is a plaintext.
```

Comment:
Does `TEXT_PLAIN` include markdown, or `(is there|should there be)` a `TEXT_MARKDOWN`?

Reply:
It includes markdown, yes, we could rename it possibly

Reply:
I think it would make more sense that way. When I'm implementing a client and I see `TEXT_PLAIN` I don't expect it to spew at me Markdown. The `PLAIN` part being key here.

Reply:
Supporting both should make most sense. Or making `TEXT_PLAIN` mandatory and `TEXT_MARKDOWN` optional.

Reply:
Not sure we want to send them both over the wire, having a textual representation makes sense (say for the chat preview), but then you get into issues about consistency (say `TEXT_MARKDOWN` is different from `TEXT_PLAIN` for malicious reasons).

So just sending over markdown and then parsing and getting a textual representation seems a bit safer, but has the drawback that is harder to gracefully introduce new markdown tags, although is not trivial in the other instance either.

For now I think we should just change `TEXT_PLAIN` to `TEXT_MARKDOWN` and handle textual representation on the receivers end.

Reply:
Fair.

Reply:
Is this part of the PR/change? Seems out of scope AFAICT.